File: app/src/main/java/com/levibostian/androidblanky/activity/BaseBackButtonActivity.java
package com.levibostian.androidblanky.activity;

import android.app.Fragment;
import android.app.FragmentTransaction;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import com.levibostian.androidblanky.R;

// Extend this class to have an activity with back button in top left corner.
public abstract class BaseBackButtonActivity extends BaseActivity {

    protected Toolbar mToolbar;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_base_back_button);

        mToolbar = (Toolbar) findViewById(R.id.back_toolbar);

        setupViews();
    }

    @Override
    protected void onResume() {
        super.onResume();

        setTitle(getScreenTitle());
    }

    protected void setFragmentInFragmentContainer(Fragment fragment) {
        getFragmentManager().beginTransaction()
                            .add(R.id.fragment_container, fragment)
                            .commit();
    }

    protected void replaceFragmentInFragmentContainer(Fragment fragment, boolean addToBackStack) {
        FragmentTransaction fragmentTransaction = getFragmentManager()
                                                          .beginTransaction()
                                                          .replace(R.id.fragment_container, fragment);

        if (addToBackStack) {
            fragmentTransaction.addToBackStack(null);
        }

        fragmentTransaction.commit();
    }

    protected abstract int getScreenTitle();

    private void setupViews() {
        setSupportActionBar(mToolbar);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                supportFinishAfterTransition();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

}

File: app/src/main/java/com/levibostian/androidblanky/activity/BaseBackButtonActivity.kt
package com.levibostian.androidblanky.activity

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentTransaction
import android.support.v7.widget.Toolbar
import android.view.MenuItem
import com.levibostian.androidblanky.R

abstract class BaseBackButtonActivity: BaseActivity() {

    protected lateinit var mToolbar: Toolbar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_base_back_button)

        mToolbar = findViewById(R.id.back_toolbar) as Toolbar

        setupViews()
    }

    override fun onResume() {
        super.onResume()

        setTitle(getScreenTitle())
    }

    protected fun setFragmentInFragmentContainer(fragment: Fragment) {
        supportFragmentManager.beginTransaction().add(R.id.fragment_container, fragment).commit()
    }

    protected fun replaceFragmentInFragmentContainer(fragment: Fragment, addToBackStack: Boolean) {
        val fragmentTransaction: FragmentTransaction = supportFragmentManager.beginTransaction().replace(R.id.fragment_container, fragment)

        if (addToBackStack) {
            fragmentTransaction.addToBackStack(null)
        }

        fragmentTransaction.commit()
    }

    protected abstract fun getScreenTitle(): Int

    private fun setupViews() {
        setSupportActionBar(mToolbar)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item!!.itemId) {
            android.R.id.home -> {
                supportFinishAfterTransition()
                return true
            }
            else -> {
                return super.onOptionsItemSelected(item)
            }
        }
    }

}